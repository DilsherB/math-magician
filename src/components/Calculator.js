import React from "react";
import Child from "./child";

const Calculator = () => {
  return (
    <div className="display">
      <div className="result">0</div>
      <div className="calculator">
        <Child />
      </div>
    </div>
  );
};

export default Calculator;
