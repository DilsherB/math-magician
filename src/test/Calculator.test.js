import renderer from "react-test-renderer";
import Calculator from "../components/Calculator";

jest.mock("../logic/calculate");

const renderTree = (tree) => renderer.create(tree);
describe("<Calculator>", () => {
  it("should render component", () => {
    expect(renderTree(<Calculator />).toJSON()).toMatchSnapshot();
  });
});
