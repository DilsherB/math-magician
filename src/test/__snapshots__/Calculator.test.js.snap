// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator> should render component 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
    0
  </div>
  <div
    className="btnsContainer"
  >
    <button
      id="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      id="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      id="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      id="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      id={7}
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      id={8}
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      id={9}
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      id="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      id={4}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      id={5}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      id={6}
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      id="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      id={1}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      id={2}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      id={3}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      id="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      id="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
